source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -493,24 +333,17 @@ void ns_dyn_mem_free(void *block)
         heap_failure(NS_DYN_MEM_DOUBLE_FREE);
     } else if (ptr < heap_main || ptr >= heap_main_end) {
         heap_failure(NS_DYN_MEM_POINTER_NOT_VALID);
-    } else if ((size < 0) || ((ptr + size) >= heap_main_end)) {
+    } else if ((ptr + size) >= heap_main_end) {
         heap_failure(NS_DYN_MEM_POINTER_NOT_VALID);
     } else {
         // Validate Sector
         if (ns_sector_validate(ptr, 1) != 0) {
             heap_failure(NS_DYN_MEM_HEAP_SECTOR_CORRUPTED);
         } else {
-            //Set Pointer to sector end len
-            //Link always prev sector
-            //Sub sequency Free
-            //Check  prev sector
-            //Try Link Next between cur
-            ns_free_and_merge_next_sector(ptr, size);
-            //Try Merge Prev with cur
-            ns_merge_prev_sector(ptr);
+            ns_free_and_merge_with_adjacent_sectors(ptr, size);
             if (mem_stat_info_ptr) {
                 //Update Free Counter
-                size = (size * sizeof(int));
+                size = ((size+2) * sizeof(int));
```

Comment:
Formatting. And excess brackets. As previous, I would prefer just using an expression to actually changing the variable.

Very hard to follow if one variable changes units or fundamental meaning throughout a function.